fenix/app/src/main/java/org/mozilla/fenix/shopping/middleware/ProductAnalysisMapper.kt

```diff
@@ -32,18 +32,15 @@ private fun ProductAnalysis.toProductReview(isInitialAnalysis: Boolean): Product
         val mappedGrade = grade?.asEnumOrDefault<ReviewQualityCheckState.Grade>()
         val mappedHighlights = highlights?.toHighlights()?.toSortedMap()
 
-        if (mappedGrade == null && mappedRating == null && mappedHighlights == null) {
-            if (isInitialAnalysis) {
-                ProductReviewState.NoAnalysisPresent()
-            } else {
-                ProductReviewState.Error.NotEnoughReviews
-            }
+        if (notEnoughReviews) {
```

Comment:
I think we can now move this condition as an else if after the `productId` null check.

Also, we could && with `needsAnalysis = false`

Reply:
Gotcha, added the check back in for when grade, rating, highlights are null